js/components/util/promises.ts

```diff
@@ -10,3 +10,22 @@ export type NoAwait<F extends (...args: unknown[]) => Promise<unknown>> =
 export function noAwait<F extends (...args: unknown[]) => Promise<unknown>>(f: F): NoAwait<F> {
     return f as NoAwait<F>;
 }
+
+/**
+ * Utility class that allows a promise to be passed to a
+ * service to be resolved.
+ */
+
+export class Ready {
+    onError: (value?: unknown) => void;
+    onReady: (value?: unknown) => void;
+
+    ready: Promise<unknown>;
+
+    constructor() {
+        this.ready = new Promise((resolve, reject) => {
+            this.onReady = resolve;
+            this.onError = reject;
+        }).catch(console.error);
```

Comment:
AFAICT, when this rejects, it's just going to log to the console. Why not add a parameter for what to call on reject? (It could default to `console.error`.)

Reply:
That seems like a good idea.